api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/AbstractConv.kt

```diff
@@ -62,40 +59,53 @@ public abstract class AbstractConv(
     protected val activityRegularizerInternal: Regularizer?,
     protected val paddingInternal: ConvPadding,
     protected val useBiasInternal: Boolean,
-    protected val kernelVariableName: String,
-    protected val biasVariableName: String,
+    protected val defaultKernelVariableName: String,
+    protected val defaultBiasVariableName: String,
     name: String
-) : Layer(name) {
+) : Layer(name), ParametrizedLayer {
+    public override val variables: List<VariableDto>
+        get() = listOfNotNull(kernel, bias)
 
     /** Tensor with kernel weights */
-    protected lateinit var kernel: Variable<Float>
+    protected lateinit var kernel: VariableDto
 
     /** Tensor with bias weights */
-    protected var bias: Variable<Float>? = null
-
-    /** Shape of internal implementation of kernel variable */
-    protected lateinit var biasShape: Shape
-
-    /** Shape of internal implementation of bias variable */
-    protected lateinit var kernelShape: Shape
+    protected var bias: VariableDto? = null
 
     override fun build(tf: Ops, kGraph: KGraph, inputShape: Shape) {
         // Amount of channels should be the last value in the inputShape
         val numberOfChannels = inputShape.size(inputShape.numDimensions() - 1)
 
-        // Compute shapes of kernel and bias matrices
-        kernelShape = computeKernelShape(numberOfChannels)
-        biasShape = computeBiasShape(numberOfChannels)
-
         // should be calculated before addWeight because it's used in calculation
         val inputDepth = numberOfChannels // number of input channels
         val outputDepth = getOutputDepth(numberOfChannels) // number of output channels
-        fanIn = (inputDepth * multiply(*kernelSizeInternal)).toInt()
-        fanOut = ((outputDepth * multiply(*kernelSizeInternal)).toDouble() /
-                multiply(*stridesInternal).toDouble()).roundToInt()
+        fanIn = (inputDepth * multiply(kernelSizeInternal)).toInt()
```

Comment:
`fanIn`/`fanOut` used only for preparing initializers, and filled only for some of the layers, maybe make sense to delete it from `Layer`